pkg/engine/resolve/resolve.go

```diff
@@ -309,7 +338,7 @@ type DataSourceBatch interface {
 }
 
 type DataSource interface {
-	Load(ctx context.Context, input []byte, w io.Writer) (err error)
+	Load(ctx *Context, input []byte, w io.Writer) (err error)
```

Comment:
You're aware that this will break all dependent DataSources?
Why is it necessary? Can we not just pass context.Context to DataSources like before?
Is there a reason why we need *resolve.Context in the Load function?

Reply:
I am aware that's an incompatible change. The reason I had to change the signature is because graphql_datasource.go requires state from the `resolve.Context` to remove the undefined variables https://github.com/wundergraph/graphql-go-tools/pull/522/files#diff-62c4a9dc18bfa9b5576bd76534407a914233956d05ed14837342302dd17168beL1411 - Can we store additional state in `input`parameter? That would allow us to carry the information about the undefined variables from here https://github.com/wundergraph/graphql-go-tools/pull/522/files#diff-b4d0466beb12697fa426019dad1c410175f54bf1466b252c6ac18033f402d266R74 and then we can pass just a `context.Context`

Reply:
I've just realized `dataLoader` also uses state from `resolve.Context` https://github.com/wundergraph/graphql-go-tools/blob/9fe3016fb1a70badd51f67b6923c4389b42a62df/pkg/engine/resolve/dataloader.go#L153 and this one would be way more complicated to refactor away

Reply:
`dataLoader` is easy to refactor because it does not need to conform to `DataSource`

Reply:
Keep in mind that other data sources, outside of our repo, might have implemented this interface.

Reply:
Since we were already changing `Context` and doing a breaking change, I assumed more breaking changes were fair game. However, thinking about this a bit more I realized we can carry the required state in the rendered data instead of passing it via the context, hence not needing a full `resolve.Context` anymore. PTAL